framework/auth/tasks.py

```diff
@@ -0,0 +1,16 @@
+from framework.celery_tasks import app
+
+@app.task
+def update_user_from_login(user_id, login_time, updates=None):
+    from osf.models import OSFUser
+    if not updates:
+        updates = {}
+    user = OSFUser.load(user_id)
+    user.update_date_last_login(login_time)
+    user.clean_email_verifications()
+    user.update_affiliated_institutions_by_email_domain()
+    if 'accepted_terms_of_service' in updates:
+        user.accepted_terms_of_service = updates['accepted_terms_of_service']
+    if 'verification_key' in updates:
+        user.verification_key = updates['verification_key']
```

Comment:
One comment, similar to `user.clean_email_verifications()`, calling `user.clean_verification_key()` may be better since setting verification key (not `None`) shouldn't never be (and is not) sent to async. However, this is out of scope of this PR and I can include it in my revised PR as mentioned.

Reply:
I like the un-keyed approach better but we can change that later.